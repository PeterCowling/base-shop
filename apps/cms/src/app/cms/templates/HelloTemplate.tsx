import React from "react";

export interface HelloTemplateProps {
  /** Main heading text */
  title: string;
  /** Body content for the template */
  body: string;
}

/**
 * Simple template rendering a heading and body copy.
 */
export function HelloTemplate({ title, body }: HelloTemplateProps) {
  return (
    <section>
      <h1>{title}</h1>
      <p>{body}</p>
    </section>
  );
}
