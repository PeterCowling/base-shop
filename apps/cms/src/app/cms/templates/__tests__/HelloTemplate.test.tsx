import { render, screen } from "@testing-library/react";
import { HelloTemplate } from "../HelloTemplate";

describe("HelloTemplate", () => {
  it("renders title and body", () => {
    const { container } = render(
      <HelloTemplate title="Welcome" body="Hello world" />
    );

    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
      "Welcome"
    );
    expect(container).toMatchSnapshot();
  });
});
