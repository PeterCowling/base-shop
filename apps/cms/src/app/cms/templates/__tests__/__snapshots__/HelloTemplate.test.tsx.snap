// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloTemplate renders title and body 1`] = `
<div>
  <section>
    <h1>
      Welcome
    </h1>
    <p>
      Hello world
    </p>
  </section>
</div>
`;
